### DISCUSSION TITLE
configuring memory

### DISCUSSION THREAD

**Discussion by budd3r** [orientechnologies/orientdb#9904] [Q&A] [answered]
Am I understanding these Orient memory settings correctly?

Total / max memory available to Orientdb:   "JVM Heap (-Xmx)"   +   "storage.diskCache.bufferSize"

The docs indicate that in large datasets, more memory should be provided to diskCache than to JVM-Heap (at a ratio of 9-to-1 per the example in the docs, although I realize that was just an example).

If running an Orient standalone server on a machine dedicated to running Orientdb with, say 64G of memory, would this be reasonable?:
- Reserve some memory for OS & misc overhead: 7G     _(not sure what is appropriate here, thinking min 2G or 10% of available)_
- JVM Heap (-Xmx): 19G                                                   _(1/3 of remaining memory)_
- storage.diskCache.bufferSize: 38G (-Dstorage.diskCache.bufferSize=38912)         _(2/3 of remaining memory)_

Thanks in advance for your advice!

**Comment by tglman** [comment, marked as answer]
Hi,

Yes your number are reasonable, also it all depends on the kind of operations you do: if you have a lot of data that need to be accessed frequently a bigger disc cache is better, if you have a lot of complex query that need to do buffering in memory ( like distinct or not indexed ordering) than a bigger heap is required.

Anyway the disk cache is just a cache, so the size of it will only change the performance and need to be at most as big as the data on the disk is big.

Regards

**Reply by budd3r** [reply]
Thank you!
